File: samples/simple-discovery-app/simple-discovery-app/Controllers/NewFileViewController.m
/*******************************************************************************
 * Copyright (c) Microsoft Open Technologies, Inc.
 * All Rights Reserved
 * See License.txt in the project root for license information.
 ******************************************************************************/

#import "NewFileViewController.h"
#import "BaseController.h"

@interface NewFileViewController ()

@end

@implementation NewFileViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)CreateFile:(id)sender {
    [BaseController getClient:^(MSOEntityContainerClient * client) {
        
        __block UIActivityIndicatorView *spinner = [BaseController getSpinner:self.view];
        
        MSOItem* item = [[MSOItem alloc] init];
        item.name = self.txtName.text;
        item.type = @"File";
       
        NSData* body = [self.txtBody.text dataUsingEncoding:NSUTF8StringEncoding];
        
        [[[[client getme] getfiles] add:item :^(MSOItem *item, NSError *e) {
            __block NSString* _id = item.id;
            
            [[[[[[client getme] getfiles] getById:_id] asFile] putContent:body :^(int result, NSError *error) {
                
                [[[[[[client getme] getfiles] getById:_id] asFile] getContent:^(NSData *content, NSError *error) {
                    
                    dispatch_async(dispatch_get_main_queue(),
                                   ^{
                                       
                                       NSString* title = @"Error!";
                                       NSString* contentResultString = [NSString alloc];
                                       
                                       if(error == nil){
                                           title = @"File Created!";
                                           contentResultString = [[NSString alloc] initWithData:content encoding:NSUTF8StringEncoding];
                                       }else{
                                           contentResultString = [[error userInfo] objectForKey:0];
                                       }
                                       
                                       [spinner stopAnimating];
                                       UIAlertView* alert = [[UIAlertView alloc] initWithTitle:title message:contentResultString delegate:self cancelButtonTitle:@"Ok" otherButtonTitles:nil, nil];
                                       [alert show];
                                   });
                    
                }] resume];
                
                
            }] resume];
        }] resume];
    }];
}
@end

File: sdk-swift/samples/simple-files-app/simple-files-app/NewFileViewController.swift
//
//  NewFileViewController.swift
//  simple-files-app
//
//  Created by Gustavo on 11/10/14.
//  Copyright (c) 2014 MsOpentech. All rights reserved.
//

import UIKit
import office365_files_sdk

class NewFileViewController: UIViewController {

    @IBAction func Create(sender: AnyObject) {
        
        var item = MSSharePointItem();
        item.name = self.txtFileName.text;
        item.type = "File";
        var data = NSData(contentsOfFile: self.txtBody.text!);
        BaseController.getClient({(client : MSSharePointClient) -> Void! in
            
            client.getFiles().add(item, callback: { (result : MSSharePointItem?, error : NSError?) -> Void! in
                var file : MSSharePointItem = result!;
                client.getFiles().getById(file.id).asFile().putContent(data!, callback: { (result, error) -> Void! in
                    
                    client.getFiles().getById(file.id).asFile().getContent({ (content, error) -> Void! in
                        dispatch_async(dispatch_get_main_queue(), { () -> Void in
                            var stringContent = NSString(data: content!, encoding: NSUTF8StringEncoding);
                            var alert = UIAlertView(title: "File Created!", message: stringContent, delegate: self, cancelButtonTitle: "Ok");
                            alert.show();
                        });
                        
                    }).resume();
                
                }).resume();
                
                return Void();
            }).resume();
        });
    }
    
    @IBOutlet weak var txtBody: UITextView!
    @IBOutlet weak var txtFileName: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
